src/__tests__/OrsSnap.cy.js

```diff
@@ -0,0 +1,103 @@
+import OrsSnap from '../OrsSnap.js'
+
+const key = Cypress.env('api_key')
+const orsSnap = new OrsSnap({ 'api_key': key })
+
+describe('Snap Test', () => {
+
+    context('construction', () => {
+        it('sets correct service and API version', () => {
+            expect(orsSnap.defaultArgs.service).to.equal('snap')
+            expect(orsSnap.defaultArgs.api_version).to.equal('v2')
+        })
+    })
+
+    context('methods', () => {
+        context('getBody', () => {
+            it('parses json.options if not an object', () => {
+                let bodyArgs = orsSnap
+                    .getBody({'options': '{ "option1": "o1", "option2": "o2" }'})
+                expect(bodyArgs['options']).to.deep.equal({ "option1": "o1", "option2": "o2" })
+            })
+
+        })
+
+        context('calculate', () => {
+            it('fails without parameters', (done) => {
+                orsSnap.calculate({})
+                    .catch((err) => {
+                        expect(err.status).to.equal(400)
+                        expect(err.message).to.equal('Bad Request')
+                        done()
+                    })
+            })
+
+            it('snap response has snapped_distance', (done) => {
+                orsSnap.calculate({
+                    locations: [[8.681495,49.51461],[8.686507,49.41943]],
+                    radius: 300,
+                    profile: 'driving-car',
+                    format: 'json'
+                }).then((json) => {
+                    expect(json['locations'].length).to.be.greaterThan(0)
+                    expect(json['locations'][1].snapped_distance).exist
```

Comment:
existance of snapped distance can be tested together with the simple snap result.
I would rather have another test for not found entries (null,[location])